src/pages/payments/create/Create.tsx

```diff
@@ -128,6 +128,31 @@ export function Create() {
     ]);
   };
 
+  const handleInvoiceReplace = (
+    newInvoice: {
+      id: string;
+      amount: number;
+      balance: number;
+    },
+    oldInvoiceId: string
+  ) => {
+    const item = formik.values.invoices.find(
+      (invoice: { _id: string; amount: string; invoice_id: string }) =>
+        invoice.invoice_id === oldInvoiceId
+    );
+    if (item)
+      formik.setFieldValue(
+        `invoices[${formik.values.invoices.indexOf(item)}]`,
+        {
+          _id: v4(),
+          amount:
+            newInvoice.balance > 0 ? newInvoice.balance : newInvoice.amount,
+          credit_id: '',
+          invoice_id: newInvoice.id,
+        }
+      );
+  };
+
```

Comment:
I assume this is related to the filtering itself. Not sure why we do this, maybe we should communicate about this. It's hard to follow it this way.